application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/TeXCommand.java

```diff
@@ -57,21 +64,28 @@ public void onSlashCommand(@NotNull final SlashCommandEvent event) {
         String userID = (Objects.requireNonNull(event.getMember()).getId());
         TeXFormula formula;
         try {
+            if (latex.contains("$")) {
+                latex = convertInlineLatexToFull(latex);
+            }
             formula = new TeXFormula(latex);
         } catch (ParseException e) {
             event.reply("That is an invalid latex: " + e.getMessage()).setEphemeral(true).queue();
             return;
         }
         event.deferReply().queue();
-        Image image = formula.createBufferedImage(TeXConstants.STYLE_DISPLAY, DEFAULT_IMAGE_SIZE,
-                FOREGROUND_COLOR, BACKGROUND_COLOR);
-        if (image.getWidth(null) == -1 || image.getHeight(null) == -1) {
-            event.getHook().setEphemeral(true).editOriginal(RENDERING_ERROR).queue();
-            logger.warn(
-                    "Unable to render latex, image does not have an accessible width or height. Formula was {}",
-                    latex);
+        Image image = getImage(event, latex, formula);
+        ByteArrayOutputStream renderedTextImageStream =
+                getRenderedTextImageStream(event, latex, image);
+        if (renderedTextImageStream == null)
             return;
-        }
+        sendImage(event, userID, renderedTextImageStream);
+    }
+
+    @Nullable
+    private ByteArrayOutputStream getRenderedTextImageStream(@NotNull SlashCommandEvent event,
```

Comment:
I dont like the name. You are using `get` as if it would be an innocant getter - but it is not. It is actually doing some heavy stuff.

The method converts an `Image` into `ByteArrayOutputStream`. So maybe `convertImageToRaw` would be more appropriate.